src/figuresClasses.ts

```diff
@@ -1,19 +1,95 @@
-export interface Figure {
-
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
 }
 
-export class Triangle {
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue'
+}
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (
+      a <= 0
+      || b <= 0
+      || c <= 0
+    ) {
+      throw new Error('Every side should be a positive number');
+    } else if (
+      a >= (b + c)
+      || b >= (a + c)
+      || c >= (a + b)
+    ) {
+      throw new Error('The longest side is longer than sum of two others');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+
+    const areaOfTriangle = s * (s - this.a) * (s - this.b) * (s - this.c);
 
+    const result = Math.sqrt(areaOfTriangle);
+
+    return Math.floor(result * 100) / 100;
```

Comment:
Optional: you round with the same logic twice over the code. You also could create a separate helper function - `rounder` for example for this purpose.